### PR TITLE
WHO - Doctor's companion and support

### LINKED ISSUE #3504: DoctorWho: Doctor's Companion
> You can have two commanders if the other is the Doctor.

Partner variant

### PR DESCRIPTION
**Pull request by Northmoc**
- closes #3504

### CODE REVIEW

**Review by Hanmac — APPROVED** [review]

File: forge-gui/res/cardsfolder/upcoming/donna_noble.txt
Comment (Glorax) [inline forge-gui/res/cardsfolder/upcoming/donna_noble.txt:1]: "TrigToken" needs to be chanced to "TrigDamage". The script doesn't have any "TrigToken" to reference, but it does have "TrigDamage".

Comment (Glorax) [reply forge-gui/res/cardsfolder/upcoming/donna_noble.txt:1]: Actually testing the card reveals that the fix appears functional. It seems to be acting as intended.

File: forge-core/src/main/java/forge/card/CardRules.java [hunk 1]
```diff
@@ -275,18 +275,53 @@ public boolean canBeCommander() {
...
+        }
+        if (hasKeyword("Friends forever") && b.hasKeyword("Friends forever")) {
+            legal = true; // Stranger Things Secret Lair gimmick partner commander
+        }
+        if (hasKeyword("Choose a Background") && b.canBeBackground()
+                || b.hasKeyword("Choose a Background") && canBeBackground()) {
+            legal = true; // commander with background
+        }
+        if (isDoctor() && b.hasKeyword("Doctor's companion")
+                || hasKeyword("Doctor's companion") && b.isDoctor()) {
```
Comment (kevlahnota) [inline forge-core/src/main/java/forge/card/CardRules.java:297]: IIRC @tool4ever updated to use `haskeyword(Keyword.key)`, maybe just use `hasKeyword(Keyword.DOCTORS_COMPANION)` and maybe update others too..

Comment (Hanmac) [reply forge-core/src/main/java/forge/card/CardRules.java:297]: Doesn't work
Only Card has `hasKeyword(KeywordEnum)`

CardRules only has `hasKeyword(String)`

Also the KeywordEnum only exist in Forge-Game, not Forge-Core

Comment (Hanmac) [reply forge-core/src/main/java/forge/card/CardRules.java:297]: @tehdiplomat we need to think about it if we really still need Forge-Core or if we should merge this with Forge-Game

File: forge-gui/res/cardsfolder/upcoming/sarah_jane_smith.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Name:Sarah Jane Smith
+ManaCost:1 W
+Types:Legendary Creature Human Detective
+PT:2/1
+T:Mode$ SpellCast | ValidCard$ Card.Historic | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigInvestigate | TriggerDescription$ Whenever you cast a historic spell, investigate. This ability triggers only once each turn. (Artifacts, legendaries, and Sagas are historic. To investigate, create a Clue token. It's an artifact with "{2}, Sacrifice this artifact: Draw a card.")
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/sarah_jane_smith.txt:5]: "Only once" not handled

**Review by Hanmac — APPROVED** [review]

**Pull request merged**